pkg/storage/test/authz_models.go

```diff
@@ -161,3 +163,43 @@ func FindLatestAuthorizationModelIDTest(t *testing.T, datastore storage.OpenFGAD
 		require.Equal(t, newModel.Id, latestID)
 	})
 }
+
+func ReadAuthorizationModelUnmarshallErrorTest(t *testing.T, engine string, uri string, ds storage.OpenFGADatastore) {
```

Comment:
Up until now the abstraction has been to pass any implementation of the OpenFGADatastore interface to these test methods. The reason for that is so that we can run the storage conformance tests against anything implementing the interface.

I originally said that it would be nice to remove the duplicate code in postgres and mysql, but now I'm wondering if removing the duplicate code is just leaving other non-idealisms behind. It'd be good to avoid implementation specific code in the general storage conformance test suite.

I'm wondering if there is a way we can get the best of both worlds (avoid the implementation specific behavior in the general storage test suite AND not have the duplicate code) 🤔 . Do you have any ideas? If not, should we revert to what you originally had (I'm sorry 😄 )?